Runtime/src/main/java/com/rakuten/tech/mobile/perf/runtime/internal/Util.java

```diff
@@ -63,4 +73,48 @@
       return null;
     }
   }
+
+  /**
+   * Check if the application is debuggable.
+   *
+   * This method check if the application is signed by a debug key.
+   * https://stackoverflow.com/questions/7085644/how-to-check-if-apk-is-signed-or-debug-build
+   *
+   * @param context application context
+   * @return true if app is debuggable, false otherwise
+   */
+  static boolean isAppDebuggable(Context context)
+  {
+    boolean debuggable = false;
+
+    try
+    {
+      PackageInfo pinfo = context.getPackageManager().getPackageInfo(context.getPackageName(), PackageManager.GET_SIGNATURES);
+      Signature signatures[] = pinfo.signatures;
```

Comment:
from the [docs](https://developer.android.com/reference/android/content/pm/PackageInfo.html#signatures)

> Array of all signatures read from the package file. This is only filled in if the flag GET_SIGNATURES was set. A package must be singed with at least one certificate which is at position zero. The package can be signed with additional certificates which appear as subsequent entries. Note: Signature ordering is not guaranteed to be stable which means that a package signed with certificates A and B is equivalent to being signed with certificates B and A. This means that in case multiple signatures are reported you cannot assume the one at the first position to be the same across updates.

is there a valid case where a production app will have multiple signatures and one of them being a debug principle? I can't think of a case, but just want to be sure.

Reply:
`Google Play does not accept APKs with multiple signatures.` mentioned here:
https://stackoverflow.com/questions/2881904/how-to-sign-an-apk-with-more-than-one-certificate